### PR TITLE
Add component-library analytics documentation

### PR DESCRIPTION
**Pull request by bsmartin-ep**
Document....

- [x] component-library level analytics
- [x] _va.gov_ static-page application level analytics 
- [x] future with web components
- [x] application level analytics
- [ ] _platform/analytics/google-analytics/tracking-design-system-components.md_ cleanup

### CODE REVIEW

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 1]
```diff
@@ -44,25 +44,68 @@ To support our React-based single page apps, we have turned on reporting of Hist
...
+We also send custom event tracking via JavaScript to report on user actions like downloads, successful or failed form submissions, _etc_. Since Event tracking is more customized than the pageview tracking, it relies on defining and instrumenting a dataLayer event. We are also able to send other non-PII/PHI values like `errorKeys` into Google Analytics for further measurement.
 
-During the Discovery phase, your VFS team should have worked with Analytics-Insights to define the key performance indicators you'd like to track. In the build phase, we will work with you to define these requirements for Event Tracking. 
+During the Discovery phase, your VFS team should have worked with Analytics-Insights to define the key performance indicators you'd like to track. In the build phase, we will work with you to define these requirements for Event Tracking. The good news is that many interactions automatically generate Google Analytics events:
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:51]: 🎉 Nice addition here

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 2]
```diff
@@ -44,25 +44,68 @@ To support our React-based single page apps, we have turned on reporting of Hist
...
-_Instrumentation:_
+For example, the AlertBox component calls the dispatchAnalyticsEvent helper function, which emits a CustomEvent on the document.body element.   The listener on va.gov matches the componentName and action in the analyticsEvents lookup table to get an event like  'nav-alert-box-link-click'.  The rest of the fields in the CustomEvent's details are given a prefix of the componentName (kebab format) and sent along with the event. 
+
+##### Other important things:
+
+*   Some components are opt-out and you need to add a disableAnalytics param to the component to disable analytics.
+*   Others (specifically form controls) are opt-in.  Due to the risk of PII/PHI, you must add an enableAnalytics flag to these components to get the CustomEvent to emit.
+*   We try to match the names of the additional params to their component counterparts.  this.props.label would be label while this.props.options label would be optionLabel.
+*   There's a lot of boilerplate in the unit tests for these in the component-library, so use the helper function if possible.
+*   Helpful links:
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:66]: @bsmartin-ep This is excellent - would you mind giving this section with the important links a subheader markdown 

```suggestion
*   ### Helpful links:
```

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 3]
```diff
@@ -44,25 +44,68 @@ To support our React-based single page apps, we have turned on reporting of Hist
...
+    *   https://github.com/department-of-veterans-affairs/component-library/
+    *   https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/site-wide/component-library-analytics-setup.js
+    *   https://github.com/department-of-veterans-affairs/component-library/blob/master/src/helpers/analytics.js
+    *   https://github.com/department-of-veterans-affairs/component-library/blob/master/src/helpers/test-helpers.js
+
+
+#### Forms System
+
+There are several analytics events baked into actions in the forms-system.  If you look in the primary actions.js file, you will see dataLayer pushes (via helper function recordEvent) for submission, successful submission, and others.  Keep in mind you will need to use a product prefix we have allowed in GTM.
+
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:78]:
```suggestion
For a complete list of our existing forms library events, please see our documentation [here](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/analytics/google-analytics/tracking-form-events.md). 
```

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 4]
```diff
@@ -44,25 +44,68 @@ To support our React-based single page apps, we have turned on reporting of Hist
...
+*   We try to match the names of the additional params to their component counterparts.  this.props.label would be label while this.props.options label would be optionLabel.
+*   There's a lot of boilerplate in the unit tests for these in the component-library, so use the helper function if possible.
+*   Helpful links:
+
+    *   https://design.va.gov/components/
+    *   https://github.com/department-of-veterans-affairs/component-library/
+    *   https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/site-wide/component-library-analytics-setup.js
+    *   https://github.com/department-of-veterans-affairs/component-library/blob/master/src/helpers/analytics.js
+    *   https://github.com/department-of-veterans-affairs/component-library/blob/master/src/helpers/test-helpers.js
+
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:73]:
```suggestion
* https://github.com/department-of-veterans-affairs/va.gov-team/blob/ga-event-label-consolidation/teams/vsp/teams/insights-analytics/ga-events-data-dictionaries.md#design-system-component-tracking
```

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 5]
```diff
@@ -44,25 +44,68 @@ To support our React-based single page apps, we have turned on reporting of Hist
...
+#### Component Library
 
+The component-library is an npm package of reusable components used on va.gov.  Many of these components are programmed to emit a CustomEvent during certain interactions.  va.gov listens for these events and pushes a corresponding event to the dataLayer, which then triggers an event in Google Tag Manager.
 
-_Instrumentation:_
+For example, the AlertBox component calls the dispatchAnalyticsEvent helper function, which emits a CustomEvent on the document.body element.   The listener on va.gov matches the componentName and action in the analyticsEvents lookup table to get an event like  'nav-alert-box-link-click'.  The rest of the fields in the CustomEvent's details are given a prefix of the componentName (kebab format) and sent along with the event. 
+
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:59]: @bsmartin-ep 
```suggestion
If you're a FE developer wanting to get an idea of the data/metadata automatically collected for a specific design system component -- check out the "With Analytics" labeled components in [Storybook](design.va.gov/storybook), execute the `monitorEvents(document.body, 'component-library-analytics')` in the console and expand the `detail` properties in the `CustomEvent` object

```

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 6]
```diff
@@ -44,25 +44,68 @@ To support our React-based single page apps, we have turned on reporting of Hist
...
+#### Forms System
+
+There are several analytics events baked into actions in the forms-system.  If you look in the primary actions.js file, you will see dataLayer pushes (via helper function recordEvent) for submission, successful submission, and others.  Keep in mind you will need to use a product prefix we have allowed in GTM.
+
+
+#### Static Sites pages
+
+Static content (such as Drupal content) on va.gov is run through an application called static-pages.  Because most of these pages can not use components from the component-library, we have implemented several application-wide event listeners for things like CTA buttons, Action Links, and other.  These can be found in this directory.
+
+One important development is the transition of the component-library from React components to web components.  These new web components should be able to be used in static pages.
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:84]: Should you also touch on any of the liquid templates you updated @bsmartin-ep ? Breadcrumbs?

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 7]
```diff
@@ -76,6 +119,7 @@ The naming convention for products begins with these prefixes:
 |GIBCT| `gibct-`|
 |Health Care Applications | `hca-`, `vaos-` |
 |Health Records| `health-record-` |
+|Interaction| `int-` |
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:122]:
```suggestion
|Interactions | `int-` |
```

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md [hunk 8]
```diff
@@ -44,25 +44,68 @@ To support our React-based single page apps, we have turned on reporting of Hist
...
 
 1. Analytics-Insights will guide you in the naming convention. 
-1. FE teams will implement this into code.  There is a [helper function that can be used](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/monitoring/record-event.js).
-1. In parallel, we update Google Tag Manager to pick up these events. We define an Event Category, Event Action, and Event Label that translates the dataLayer event into a more readable format for Google Analytics users. 
-1. Once the code is on staging, we will QA that it is working correctly. 
+2. FE teams will implement this into code.  There is a [helper function that can be used](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/monitoring/record-event.js).
+3. In parallel, we update Google Tag Manager to pick up these events. We define an Event Category, Event Action, and Event Label that translates the dataLayer event into a more readable format for Google Analytics users. 
+4. Once the code is on staging, we will QA that it is working correctly. 
 
 The naming convention for products begins with these prefixes:
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:110]: @bsmartin-ep Reference and link to [this list](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md#event-tracking)

File: platform/analytics/google-analytics/tracking-form-events.md [hunk 9]
```diff
@@ -16,10 +16,10 @@ In the table where it lists \<product\>, please reference the Product prefix tab
...
 | Form Starts                | Form Start - Begin form button after successful sign in                                     | `login-successful-start-form`        |
-| Form Saves and Submissions | Form Saves - Event is triggered for auto form saves                                         | `\<product\>-sip-form-saved`          |
-| Form Saves and Submissions | Form Prefill Failures - Tracks when form prefill cannot be retrieved                        | `\<product\>-sip-form-prefill-failed` |
-| Form Saves and Submissions | Form Submission - Measure the form submission; Success and failures are measured distinctly | `\<product\>--submission`             |
-| Form Saves and Submissions | Form Submission Successful - Triggered when the form has been successfully submitted        | `\<product\>--submission-successful`  |
-| Form Saves and Submissions | Form Failures - Triggered when the form has failed client or server-side                    | `\<product\>--submission-failed`      |
+| Form Saves and Submissions | Form Saves - Event is triggered for auto form saves                                         | `<product>-sip-form-saved`          |
```
Comment (jonwehausen) [inline platform/analytics/google-analytics/tracking-form-events.md:21]: @bsmartin-ep Go ahead and add the `--submission-failed-client` event in here as well 

Could we also add in some verbiage to specifically call out the double dash `--` before `submission`?

**Review by jonwehausen — CHANGES_REQUESTED** [review]
@bsmartin-ep Overall for the most part looks great man! Following these changes we'll be good to publish this draft.

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md (hunk 6 above)
Comment (bsmartin-ep) [reply platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:85]: Great idea.  Updated,

File: platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md (hunk 8 above)
Comment (bsmartin-ep) [reply platform/analytics/google-analytics/build-process-measuring-pageviews-and-events.md:111]: Not sure I understand @jonwehausen.  You want me to anchor to somewhere on the same page?

**Review by jonwehausen — COMMENTED** [review]
@bsmartin-ep This can merge when the changes have been updated.

**Pull request merged**